src/spdx2/agent/spdx2.php

```diff
@@ -443,7 +443,8 @@ protected function toLicensesWithFiles(&$filesWithLicenses, $treeTableName)
       if (($filesProceeded&2047)==0) {
         $this->heartbeat(0);
       }
-      $fullPath = $treeDao->getFullPath($fileId,$treeTableName,0,true);
+      $parentId = $treeDao->getMinimalCoveringItem($uploadId, $treeTableName);
```

Comment:
I think the value of `$parentId` is going to be constant for each iteration of the loop. Wouldn't it be better if it is initialized outside the loop only once?